themes/terrain-shading.yaml

```diff
@@ -0,0 +1,307 @@
+            draw:
+                polygons:
+                    # reuse the inherited base color, but use modified shader
+                    # that adjusts brightness / contrast to match original color
+                    style: global.sdk_terrain_shading_bathymetry_style
+                    order: global.feature_order
+                    color:
+
+        lakes:
+            bathymetry:
+                enabled: global.sdk_bathymetry
+                draw:
+                    polygons:
+                        # reuse the inherited base color, but use modified shader
+                        # that adjusts brightness / contrast to match original color
+                        style: global.sdk_terrain_shading_bathymetry_style
+                        order: global.feature_order
+                        color:
+
+        other-water-areas:
+            bathymetry:
+                enabled: global.sdk_bathymetry
+                draw:
+                    polygons:
+                        # reuse the inherited base color, but use modified shader
+                        # that adjusts brightness / contrast to match original color
+                        style: global.sdk_terrain_shading_bathymetry_style
+                        order: global.feature_order
+                        color:
+
+    landuse:
+        national_park:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+            us_national_park:
+                draw:
+                    polygons:
+                        style: global.sdk_terrain_shading_landuse_style
+                        color:
+
+        conservation:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+
+        national_forest_level_6:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+
+        forest-landcover:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+
+        parks-and-national-forests-not-national-park:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+            national_park:
+                draw:
+                    polygons:
+                        style: global.sdk_terrain_shading_landuse_style
+                        color:
+
+        farm:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
+
+        university:
+            enabled: false
+
+        cemetery:
+            draw:
+                polygons:
+                    style: global.sdk_terrain_shading_landuse_style
+                    color:
```

Comment:
@bcamper this is valid YAML and produces the expected result (to not compete with the various `color-*` themes definition of same), but is there a better Tangram convention to use?

Reply:
I think this is fine - @matteblair this is OK right?

Reply:
Hmm.. this might be of concern on ES, we expect a `scalar`, `sequence` or a `map` of value for a style parameter.

Reply:
I'm not sure what the intended result of this is. Varun is correct that this is a `null` YAML value, which Tangram ES is effectively the same as not having a `color:` entry at all.

Reply:
As a comparison:

This is the default behavior on ES (with bath on, purple-green theme, terrain-shading):
![screen shot 2017-12-07 at 4 30 01 pm](https://user-images.githubusercontent.com/360641/33745505-0b78712a-db6c-11e7-99e5-cf87c9cd151a.png)

This is when I modify the ES source to allow for `null` values by simply setting an empty string as the value in case of a `null` value being parsed from the yaml file (which probably should not be done, to avoid unexpected behavior from a user error):
![screen shot 2017-12-07 at 4 29 16 pm](https://user-images.githubusercontent.com/360641/33745533-409684d2-db6c-11e7-9dde-edad8af7f2c4.png)

Reply:
Huh, I thought we had previously discussed that for scene merge behavior, any `null` value will simply overwrite the previous value (I may be mistaken though.. however that is what JS does!). So the effect here would be to "unset" the color. For example, it's similarly been used to "remove" an imported layer.

Now I don't actually know the details of *why* this is being done here, or what the intended result is, so there may be a different/better solution.

But we should also try to make the behavior consistent between JS/ES.

Reply:
I'm not certain that this hack is still required – previously some of the styles had been applying patterns by setting alpha only and relying on an underlying color to show through. IIRC the inherited `color` of the style had been interfering with the imported `color` from the color-* schemes, and the null `color` fixed this. However we recently reworked those styles so that they set the rgb values directly.

At any rate I just commented out those null `color` blocks and don't see any difference in the styles, so it may be we can just nix them. @sensescape, can you confirm? I'm not 100% I'm looking at the right test case…

Reply:
Yesterday I worked it out to be `color: white` which was solving things on ES. I am not sure just removing that `color` statement will resolve it for ES, will try it tough. 

With respect setting it to white works probably because its multiplied to the color computed in the shader. But i will confirm this today. Me and Geraldine confirmed it to work on JS and ES both yesterday.

Reply:
Ok I think this reasoning for `color: white` makes sense:

In the `terrain-shading` style the color variant (`v_color`) is multiplied with the calculated terrain exaggeration values. Color variant picks its values from the draw rule property, so if one has to reset the color here, it does make sense to set these to `white`.
https://github.com/tangrams/refill-style/blob/e80d3f99115a41cf2cf38fb86d9911bf18b7910d/themes/terrain-shading.yaml#L271

Reply:
>Huh, I thought we had previously discussed that for scene merge behavior, any null value will simply overwrite the previous value (I may be mistaken though.. however that is what JS does!). So the effect here would be to "unset" the color. 

@bcamper scene merging is correctly overriding and marking the final `color` value here to `null`. But we do not allow style parameter values to be `null` in ES land. Only legal `Scalar`, `Sequence` or `Map` yaml values are allowed. So during parsing of the merged scene when setting the draw rules for a scene, color values are not parsed as its overriden to be `null`, and hence not added to the set of draw rules for the layer. However JS here is interpretting the `null` color value as a legal color value and probably interpreting as `white`, do you know if thats a default color being set somewhere in JS for the default styles?

Reply:
Yes, we'll need color: white for terrrain shading, which Varun and I confirmed yesterday. And Peter and I resolved the terrain pattern alpha  issue earlier this week (where shaders were weaving with each other).

Reply:
Ok, I think I've figured out what's going on here. In both JS and ES the `color: null` values were overriding corresponding `color` values from other scene files during importing. The difference is when rules are merged for these layers. In JS the `null` is just merged with other `color` parameters. In ES the `null` is *ignored* and not merged with other parameters (like I said, it's as if the parameter weren't there). You can see this by commenting out the `color: ` lines in the theme files - it produces the same (incorrect) output in both ES and JS. 

I think JS uses the same code to merge scene files and draw parameters, so "overriding" with null would naturally work. In ES these are different code paths. We could hypothetically make this instance work the same in ES if we wanted that.

Reply:
That would be my preference since it would be difficult to branch the JS code to have slightly different behaviors based on this context.

Reply:
Hmm, if thats the case, https://github.com/tangrams/tangram-es/pull/1747 should address it. With this ES will not simply ignore a `null` style param value and will aptly work with merging of draw rules.

Also works with `color: `, and we do not need to set `color: white`, if need be.

Reply:
But probably since change in ES will go in our next release, I suggest to have the `color: white` in here, which is totally fine logically also.